Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -0,0 +1,583 @@
+{
+    public class SeriesService : ISeriesService
+    {
+        private readonly CCTContext _context;
+        private readonly IMatchService _matchService;
+
+        public SeriesService(CCTContext context, IMatchService matchService)
+        {
+            _context = context;
+            _matchService = matchService;
+        }
+        public IEnumerable<LookupViewModel> GetSeriesLookup(string type)
+        {
+            if (type == "status")
+            {
+                var res = _context.SeriesStatus
+                    .Select(s => new LookupViewModel
+                    {
+                        ID = s.ID,
+                        Description = s.Description
+                    })
+                    .AsEnumerable();
+                return res;
+            }
+            if (type == "series")
+            {
+                var res = _context.SeriesType
+                    .Select(s => new LookupViewModel
+                    {
+                        ID = s.ID,
+                        Description = s.Description
+                    })
+                    .AsEnumerable();
+                return res;
+            }
+            return null;
+        }
+        public IEnumerable<SeriesExtendedViewModel> GetSeries(bool active = false)
+        {
+            var series = _context.Series
+                            .Select(s => new SeriesExtendedViewModel
+                            {
+                                ID = s.ID,
+                                Name = s.Name,
+                                StartDate = s.StartDate,
+                                EndDate = s.EndDate,
+                                Type = _context.SeriesType
+                                            .FirstOrDefault(st => st.ID == s.TypeID)
+                                            .Description,
+                                Status = _context.SeriesStatus
+                                            .FirstOrDefault(ss => ss.ID == s.StatusID)
+                                            .Description,
+                                ActivityID = s.ActivityID,
+                                Match = _matchService.GetMatchBySeriesID(s.ID),
+                                TeamStanding = _context.SeriesTeam
+                                .Where(st => st.SeriesID == s.ID)
+                                .Select(st => new TeamRecordViewModel
+                                {
+                                    ID = st.ID,
+                                    Name = _context.Team
+                                            .FirstOrDefault(t => t.ID == st.TeamID)
+                                            .Name,
+                                    Win = st.Win,
+                                    Loss = st.Loss,
+                                    Tie = _context.SeriesTeam
+                                            .Where(total => total.TeamID == st.TeamID && total.SeriesID == s.ID)
```

Comment:
What is `total`? I would think you would want something like `st` since you are filtering all SeriesTeams

Reply:
was an early implementation and st was taken. updated name to `_st`